packages/ckeditor5-emoji/src/emojimention.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module emoji/emojimention
+ */
+
+import { Database } from 'emoji-picker-element';
+import { logWarning, type LocaleTranslate } from 'ckeditor5/src/utils.js';
+import { Plugin, type Editor } from 'ckeditor5/src/core.js';
+import EmojiPicker from './emojipicker.js';
+import type { NativeEmoji } from 'emoji-picker-element/shared.d.ts';
+
+import {
+	type MentionFeed,
+	type MentionFeedObjectItem
+} from '@ckeditor/ckeditor5-mention';
+
+const EMOJI_PREFIX = 'emoji';
+const SHOW_ALL_EMOJI = '__SHOW_ALL_EMOJI__';
+
+/**
+ * The emoji mention plugin.
+ *
+ * Introduces the autocomplete of emojis while typing.
+ */
+export default class EmojiMention extends Plugin {
```

Comment:
See the order of methods in https://github.com/ckeditor/ckeditor5/blob/71c7bf3d62962b0899706dad787764319c4d5193/packages/ckeditor5-autosave/src/autosave.ts#L55.

The static getters are defined before constructor.